runtime/native.go

```diff
@@ -338,6 +338,13 @@ func getNativeType(rtype reflect.Type) *Type {
 			}
 		}
 		t = &newNativeType(rtype, base, newStringDict(d)).Type
+		t.slots.Native = &nativeSlot{func(f *Frame, o *Object) (reflect.Value, *BaseException) {
+			val, raised := base.slots.Native.Fn(f, o)
+			if raised != nil {
+				return reflect.Value{}, raised
+			}
+			return val.Convert(toNativeMetaclassUnsafe(o.typ.ToObject()).rtype), nil
+		}}
```

Comment:
It'll override the method of the primitive base type for the subclass t. The override calls the base implementation to get the primitive object and then coverts it to the (compatible) subtype.

E.g. `type devNull int` produces the Python `<type devNull>`. Without the override, the native object returned from a `<type devNull>` is the plain Go int type. The override converts this to the Go devNull type. From the original issue, devNull satisfies the `Writer` interface, whereas int does not.